### DISCUSSION TITLE
求助：执行import lava.lib.dl.slayer失败，疑似lava.lib.dl.slayer缺失

### DISCUSSION THREAD

**Discussion by zzaa888** [fangwei123456/SpikingFlow#239] [Q&A] [answered]
您好，我在pip完毕spikingjelly后，import任何activation_based里面的包总会显示报错定位至lava_exchange.py文件中的import lava.lib.dl.slayer语句。

报错显示为：
AttributeError                            Traceback (most recent call last)
<ipython-input-6-f33e041fe171> in <module>
----> 1 from spikingjelly.activation_based import neuron, encoding, functional, surrogate, layer
~\Desktop\spikingjelly-master\spikingjelly-master\spikingjelly\activation_based\neuron.py in <module>
      3 import torch
      4 import torch.nn as nn
----> 5 from . import surrogate, base, lava_exchange
      6 from .. import configure
      7 import math
~\Desktop\spikingjelly-master\spikingjelly-master\spikingjelly\activation_based\lava_exchange.py in <module>
     11     slayer = None
     12 
---> 13 class FasterLavaDense(slayer.synapse.Dense):
     14     @staticmethod
     15     def convert_from(slayer_dense: slayer.synapse.Dense):

AttributeError: 'NoneType' object has no attribute 'synapse'

我查阅lava_exchange.py后发现疑似在执行import lava.lib.dl.slayer as slayer语句时失败，导致cannot find reference ‘synapse’ in ' None'
请问这种情况该如何解决呢？感激不尽！
![1](https://user-images.githubusercontent.com/108392040/176383212-7c3de10f-6824-4ff7-a6d4-edebdfb5025a.jpg)
![2](https://user-images.githubusercontent.com/108392040/176383590-0b2e8b72-355e-45ec-93da-27df29fdfa18.jpg)

**Comment by fangwei123456** [comment, marked as answer]
是从源代码安装的？用git pull更新代码后再试试

**Reply by zzaa888** [reply]
重置了环境后，又重新clone了一遍恢复正常，谢谢您的答复！
